### PR TITLE
Adds function to create screenshots

### PR DESCRIPTION
**Pull request by halemmerich**
This allows the apploader to get a screenshot via `g.dump()`.

### CODE REVIEW

File: js/index.js [hunk 1]
```diff
@@ -1227,3 +1227,47 @@ if (btn) btn.addEventListener("click", event => {
...
+      }).finally(()=>{
+        Progress.hide({sticky:true});
+      });
+    }
+    oImage.src = s.split("\n")[0];
+    Progress.hide({sticky:true});
+    Progress.show({title:"Screenshot done",percent:85,sticky:true});
+
+  }, err=>{
+    showToast("Error creating screenshot: "+err,"error");
```
Comment (bobrippling) [inline js/index.js:1271]: Nice! Does this need a `Progress.hide(...)` call here?

Comment (halemmerich) [reply js/index.js:1271]: Do you mean line 1267? Without that it does not change from animated progress bar to fixed percentages. There might be a better way to do this.

**Comment by gfwilliams** [comment]
That's a great idea - thanks! Please could you add a:

```
if (device.id=="BANGLEJS") show_a_prompt("Screenshots aren't supported on Bangle.js 1")?
```

It might be a bit confusing for older Bangle.js owners otherwise

**Comment by halemmerich** [comment]
Do we want to check for the Bangle v1 in this code or would it maybe be cleaner to do that in `BangleApps` and just hide the screenshot button for v1? Are there other Espruino devices capable of screenshots using g.dump()?

**Comment by gfwilliams** [comment]
I thought showing/hiding the button might just be extra work. Most users now will be Bangle.js 2, the dialog is just to stop the inevitable complaints :)

We only support the 2 bangles, but if there is a bangle 3 it'll probably support screenshots :) I think if EspruinoApps is used with a Pixl.js then you might be able to screenshot that

File: js/index.js (hunk 1 above)
Comment (bobrippling) [reply js/index.js:1271]: Oh no, I was thinking if we hit an error we could've done `Progress.show({title:"Creating screenshot",interval:10,percent:"animate",sticky:true});`, and then no other change to `Progress` in this error handler section, if you see what I mean?

**Comment by gfwilliams** [comment]
Thanks!

**Pull request merged**